crates/bench/src/lib.rs

```diff
@@ -0,0 +1,120 @@
+use core::future::Future;
+use log::error;
+use std::cmp;
+pub use wasm_bench_macro::wasm_bench;
+
+fn now_nanos() -> u64 {
+    let window = web_sys::window().expect("should have a window in this context");
+    let performance = window
+        .performance()
+        .expect("performance should be available");
+    return (performance.now() * 1e6) as u64;
+}
+
+pub struct Bench {
+    iterations: u64,
+    ns_start: u64, // Non-zero iff timer is currently running.
+    ns_elapsed: u64,
+    pub bytes: u64,
+}
+
+impl Bench {
+    fn new() -> Bench {
+        Bench {
+            iterations: 0,
+            ns_start: 0,
+            ns_elapsed: 0,
+            bytes: 0,
+        }
+    }
+
+    pub fn iterations(&self) -> u64 {
```

Comment:
I always wonder if, stylistically or practically, we need getters like this. The only way one could change this value is if they have mutable access to it. Is there a difference between sites that have mutable access and sites which have access to the struct field directly?

Reply:
The benchmarks are passed an `&mut Bench` so they can set `b.bytes` or call `b.stop_timer()` on it, so anywhere that can read this value has permission to modify it.

In terms of the other accessors here, they also exist in `libtest`. This one is new (`libtest` hides this behind an `.iter()` function that takes a closure), but I think it appropriate to keep to make the value read-only.

I agree with the general sentiment to hide fields or depend on the struct refs being immutable rather than adding an accessor unnecessarily.